### ISSUE TITLE
[Android] When selecting image from gallery or taking photo, image loses orientation

### ISSUE THREAD

**Issue by joeferraro** [ivpusic/react-native-image-crop-picker#379]
### Version
Tell us which versions you are using: 

- react-native-image-crop-picker v0.14.1
- react-native v0.43.1

### Platform

- Android

### Expected behaviour

- Take photo or select photo from gallery
- Image is saved locally **with** original orientation

### Actual behaviour

- Take photo or select photo from gallery
- Image is saved locally **without** the original orientation

### Steps to reproduce

```
ImagePicker.openPicker({
  cropping: false,
  includeBase64: true,
  compressImageQuality: 0.5,
});
```

**Important** This only occurs when `cropping` is set to `false` and `compressImageQuality` is set to a value less than `1`.

**Comment by ivpusic** [comment]
do you have some time to take a look into this?

**Comment by joeferraro** [comment]
@ivpusic this issue seems related to [Compressor](https://github.com/zetbaitsu/Compressor). [`v2.1.0`](https://github.com/zetbaitsu/Compressor/commit/d69833452767c239f6f5b29a2dc0aca080cf11b2) of Compressor attempts to resolve the issue but there seems to be a regression for compressed images taken by the camera [as I've noted here](https://github.com/zetbaitsu/Compressor/issues/64#issuecomment-316181036).

**Comment by ferminmoli** [comment]
Is there any news about this? Image is loosing it's orientation

**Comment by ferminmoli** [comment]
I updated to 0.16.0 version (The last one) and Taking photo from camera is crashing on Android. Please, I need help with this.

**Comment by WLYau** [comment]
Hi, I'm experiencing the same issue, have updated to the latest version and the image still gets rotated? Anymore news on this?

We're using react-native version 0.39.2

**Comment by dinatazy** [comment]
Happens also with me on IOS. Image is rotated upside down. Any updates on this issue?

**Comment by Clowning** [comment]
Any updates ?

**Comment by dinatazy** [comment]
@Clowning   It turns out it's not the library's issue.  my solution was that I get the Exif data by setting property includeExif to true.  the exif object includes an orientation number. Each number indicates a specific orientation so , based on the orientation number you can rotate the image to its original position again.

**Comment by isaachinman** [comment]
@dinatazy I just hit this issue myself. Strange that more people haven't encountered it. When an image has this `Orientation` EXIF field set, `react-native-image-crop-picker` will actually swap the image dimensions, but leaves the rotation itself up to the end user.

I would expect `react-native-image-crop-picker` to leave the width and height values alone, and leave all manipulation to the discretion of the end user.

Thoughts?

**Comment by isaachinman** [comment]
If anyone else is affected by this, the fix in my case was to simply ignore the dimensions that `react-native-image-crop-picker` provides and just use RN natively:

    Image.getSize(image.path, (width, height) =>

**Comment by Jyrno42** [comment]
Having a similar issue. If anyone got a fork or an example how they got it working i would love to see it.

**Comment by drewandre** [comment]
Also seeing this issue, and it seems to be directly related to this issue for react-native-image-picker: https://github.com/react-community/react-native-image-picker/issues/655

**Comment by bonesyblue** [comment]
I've created a PR with a fix for this issue, feel free to check it out and leave feedback 👍

**Comment by ivpusic** [comment]
should be fixed in https://github.com/ivpusic/react-native-image-crop-picker/releases/tag/v0.24.0

**Comment by ahartzog** [comment]
Super excited for this fix! What's the usual release timeline to NPM? Trying to decide if I should install from the 0.24.0 release directly or wait for npm.

**Comment by ivpusic** [comment]
published now

**Comment by CrisHemswerthThor** [comment]
I have installed version 0.24.0 but facing the same image rotation issue .Please provide me some solution.

**My code is given below:**

`//function defined for handling the onPress event of Camera 
  _openCamera() {
    ImagePicker.openCamera({
      compressImageMaxHeight:windowH*0.60,
      mediaType: 'photo',
    }).then(image => {
      let photoPathFromCamera = image.path.replace('file://', '')
      //image coming from camera opens in RNPhotoEditor
      RNPhotoEditor.Edit({
        path: photoPathFromCamera,
        hiddenControls: ['crop']
      });
    });
  }`

Version:
react-native-image-crop-picker v0.24.0
react-native v0.59.3

Issue:
In my case ,the image does't rotate in debug mode but in case of release apk the image rotates automatically with the same code and I don't know why this is happening . It's really frustrating .

**Comment by twelve17** [comment]
@ivpusic - thanks for working on the fix for this issue.  I looked over the updated code (https://github.com/ivpusic/react-native-image-crop-picker/blob/b3606ea940cc02d2eab4f0a33e6ae4b0f6c13fdc/android/src/main/java/com/reactnative/ivpusic/imagepicker/Compression.java#L40).  I _might_ be misreading it, but is my understanding correct that the code is:

1. obtaining the rotation data from EXIF
2. applying the rotation to the image when resizing

In other words, is the compression step "resolving" the orientation of the image?  If so, should it remove the orientation info now that it's been applied?  Otherwise a future consumer of the image might think it needs to be rotated again?

**Comment by dphaener** [comment]
To add to @twelve17 comment above, version 0.24.0 is not fixing this for us. We are still seeing this issue with RN v0.57.8.

**Comment by karltaylor** [comment]
Also getting this issue on version `0.25.0`

```bash
# react-native info
System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz
    Memory: 61.71 MB / 16.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 12.4.0 - /usr/local/bin/node
    Yarn: 1.17.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5692245
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.4 => 0.60.4
```

**Comment by cayasso** [comment]
Can this issue be reopen, I am having this same issue in the latest version (`0.26.1`).

**Comment by BogdanRad** [comment]
I have some problem on 0.31.1 ....

**Comment by thiago-sandes** [comment]
Some problem on 0.32.2. #IOS

**Comment by nitin-gwl** [comment]
Same problems with 0.32.2. #Android Version 9 (Samsung Galaxy S8).

**Comment by namansukhwani** [comment]
I am facing the same issue in open camera images only on android image from open camera is rotated by 90 degrees by itself.
do any one have
 a solution ???

**Comment by eubnara** [comment]
How about trying option `compressImageQuality: 0.99`?
It removes `exif` metadata by compressing but rotates correctly.

I experience wrong rotation problem but I finally know that the problem comes from others.

I use `react-native-image-resizer`. Even though I set `keepMeta: true`, it loses orientation.
(it is explained in https://github.com/bamlab/react-native-image-resizer#promise-createresizedimagepath-maxwidth-maxheight-compressformat-quality-rotation--0-outputpath)

Anyway, maybe your image may go through some kind of compression or else and loses `exif` metadata and it makes your image seen in a wrong way.
So I just use `compressImageQuality: 0.99` and make `exif` metadata removed in advance.

**Comment by ngocht143** [comment]
Yours worked for me 👍

**Comment by 15110011** [comment]
Taking a photo and then it is still rotated on the newest version
0.35.2

**Comment by nmhoan76** [comment]
Still for version 0.36.. @ivpusic  Please check it

**Comment by mojtabaasadi** [comment]
Applied @eubnara solution, problem still  persists on  `0.37.2`

**Comment by developer-appdam** [comment]
I tried @eubnara solution but my images are still rotated. I have no clue why but they just keep rotating. Any tip?

**Comment by subhadeepquantiantech** [comment]
any update?

**Comment by gkasireddy202** [comment]
Same problem on v0.40.0.

**Comment by gkasireddy202** [comment]
@karltaylor - This below code works in v0.40.0?

> Also seeing this issue, and it seems to be directly related to this issue for react-native-image-picker: [react-native-image-picker/react-native-image-picker#655](https://github.com/react-native-image-picker/react-native-image-picker/issues/655)
> 
> More info about the exif orientation data that is returned ("0", "1", etc) can be found here: https://developer.android.com/reference/android/support/media/ExifInterface.html#TAG_ORIENTATION
> 
> I decided to modify the width and height based on the orientation data:
> 
> ```
> .then(photo => {
>   if (Platform.OS === 'android') {
>     let { width, height } = this.handlePhotoOrientation(photo)
>     photo.width = height
>     photo.height = width
>   }
> })
> ```
> 
> ```
> handlePhotoOrientation = photo => {
>   let photoOrientationData = photo.exif.Orientation
>   let width, height
>   if (photoOrientationData) {
>     switch (parseInt(photoOrientationData)) {
>       case 1: // ORIENTATION_NORMAL (values should be swapped)
>         return {
>           width: height,
>           height: width
>          }
>     case 3: // ORIENTATION_ROTATE_180 (values should be swapped)
>       return {
>         width: height,
>         height: width
>        }
>      default:
>         return {
>           width: photo.width,
>           height: photo.height
>         }
>     }
>   } else {
>     return {
>       width: photo.width,
>       height: photo.height
>     }
>   }
> }
> ```

**Comment by gkasireddy202** [comment]
@dphaener  - Not getting photo.exif.Orientation when I captured images from the camera.

Getting below JSON object: 

{"height": 3000, "mime": "image/jpeg", "modificationDate": "1694499770000", "path": "file:///storage/emulated/0/Android/data/com.myapp/files/Pictures/image-a92fa6d9-5308-4316-a9c5-026ee0c83f8c8876211762320268300.jpg", "size": 5275914, "width": 4000}

**Comment by gkasireddy202** [comment]
I rotated the image rotation 0 using npm:react-native-image-resizer before uploading it to the server.
works fine.
